### PR TITLE
3.0.0-SideNav

### PR DESCRIPTION
**Pull request by CloudPower97**
# Description

This Pull Request is for #605 

Dropping 'jQuery' support, now using `M.Sidenav.init()`.

Adding new options (as per the [documentation](https://materializecss.com/sidenav.html#options)):

-  inDuration
- outDuration
- onOpenStart
- onOpenEnd
- onCloseStart
- onCloseEnd
- preventScrolling

Removing deprecated options:

- menuWidth
- closeOnClick

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

I've generated new snapshot and used the new mocker.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by alextrastero** [comment]
Fixes #605

File: src/SideNav.js [hunk 1]
```diff
@@ -10,8 +10,14 @@ class SideNav extends Component {
   }
 
   componentDidMount() {
-    const { options = {} } = this.props;
-    $('.sidenav').sidenav(options);
+    if (M) {
```
Comment (alextrastero) [inline src/SideNav.js]: `typeof`

**Comment by CloudPower97** [comment]
@alextrastero ping :smile:

File: test/SideNav.spec.js [hunk 2]
```diff
@@ -26,71 +26,92 @@ function setup(props = {}, children, mounted) {
...
+      const children = sideNav.find('.test-child');
+
+      expect(children).toHaveLength(1);
     });
 
-    expect(sideNavProps.trigger).toBeUndefined();
-    expect(sideNavProps.options).toBeUndefined();
-    expect(sideNavProps.shouldTransfer).toBeTruthy();
-  });
+    test('should consume props it uses', () => {
```
Comment (alextrastero) [inline test/SideNav.spec.js]: I don't understand this spec, what do you mean by consume the props it uses?

File: test/SideNav.spec.js [hunk 3]
```diff
@@ -26,71 +26,92 @@ function setup(props = {}, children, mounted) {
...
+    });
 
-  test('should render children', () => {
-    const { sideNav } = setup({}, <span className="test-child" />);
-    const children = sideNav.find('.test-child');
+    test('should be fixed if no trigger is passed', () => {
+      const component = <SideNav className="red" />;
+      const wrapper = shallow(component);
+      const sideNav = wrapper.find('.sidenav');
+      expect(sideNav).toMatchSnapshot();
```
Comment (alextrastero) [inline test/SideNav.spec.js]: can we assert that the `fixed` class exists?

File: test/SideNav.spec.js [hunk 4]
```diff
@@ -26,71 +26,92 @@ function setup(props = {}, children, mounted) {
...
 
-  test('should render a given id', () => {
-    const { sideNavProps, triggerProps } = setup({ id: 'test' });
-    expect(sideNavProps.id).toEqual('test');
-    expect(triggerProps['data-target']).toEqual('test');
-  });
+    test('should render trigger on large screens not fixed`', () => {
+      const { trigger } = setup({ className: 'green' });
+      expect(trigger).toMatchSnapshot();
+    });
```
Comment (alextrastero) [inline test/SideNav.spec.js]: is this test useful? how to check it's a large screen?

File: test/SideNav.spec.js [hunk 5]
```diff
@@ -26,71 +26,92 @@ function setup(props = {}, children, mounted) {
...
 
-  test('should be fixed if no trigger is passed', () => {
-    const component = <SideNav className="red" />;
-    const wrapper = shallow(component);
-    const sideNav = wrapper.find('.sidenav');
-    expect(sideNav).toMatchSnapshot();
-  });
+    test('should render fixed if not passed trigger or fixed', () => {
+      const { wrapper } = setup({ className: 'red' });
+      expect(wrapper).toMatchSnapshot();
```
Comment (alextrastero) [inline test/SideNav.spec.js]: can we add specific assertions here for classes?

File: test/SideNav.spec.js (hunk 3 above)
Comment (CloudPower97) [reply test/SideNav.spec.js]: Done :+1:

File: test/SideNav.spec.js (hunk 2 above)
Comment (CloudPower97) [reply test/SideNav.spec.js]: Not sure, I didn't write this test, it was already there

File: test/SideNav.spec.js (hunk 4 above)
Comment (CloudPower97) [reply test/SideNav.spec.js]: Dropped it

**Comment by CloudPower97** [comment]
@alextrastero ping :smiley:

File: test/SideNav.spec.js [hunk 6]
```diff
@@ -65,7 +99,6 @@ describe('<SideNav />', () => {
     expect(sideNavProps.id).toEqual('test');
     expect(triggerProps['data-target']).toEqual('test');
   });
-
```
[the remark this answers is not in the document]
Comment (CloudPower97) [reply test/SideNav.spec.js]: @alextrastero It should be ok now :smile:

**Comment by alextrastero** [comment]
Nice job @CloudPower97 !

**Pull request merged**